### PR TITLE
Refactor Overlay BGP Peerings

### LINKED ISSUE #373: MPBGP remove AVD default of no neighbor for peers/peer groups in AF
**Issue Type**

- Template enhancement

**Is your feature request related to a problem? Please describe.**
Checking the EOS CLI standard behaviour and AVD there is redundant knobs for MPBGP where we explicitly disable peers/peer groups that are not meant to be part of the particular address-family.

By default with MPBGP we will not negotiate capability for any address-family that the peer is not defined, thus there is no need for us to explicitly do this so.

AVD

```
   address-family evpn
      neighbor EVPN-OVERLAY-PEERS activate
      no neighbor IPv4-UNDERLAY-PEERS activate
      no neighbor MLAG-IPv4-UNDERLAY-PEER activate
   !
   address-family ipv4
      no neighbor EVPN-OVERLAY-PEERS activate
      neighbor IPv4-UNDERLAY-PEERS activate
      neighbor MLAG-IPv4-UNDERLAY-PEER activate

```
**Describe the solution you'd like**

The following  provides the exact same result

```
PE1(config)#show run sec bgp
router bgp 1
   no bgp default ipv4-unicast
   neighbor EVPN peer group
   neighbor EVPN remote-as 2
   neighbor EVPN send-community
   neighbor EVPN maximum-routes 12000
   neighbor IP peer group
   neighbor IP remote-as 2
   neighbor IP update-source Loopback0
   neighbor IP ebgp-multihop 2
   neighbor IP maximum-routes 12000
   neighbor 2.2.2.2 peer group IP
   neighbor 10.0.0.1 peer group EVPN
   !
   address-family evpn
      bgp next-hop-unchanged
      neighbor EVPN activate
   !
   address-family ipv4
      bgp next-hop-unchanged
      neighbor IP activate
```
```
PE1(config)#show ip bgp summ
BGP summary information for VRF default
Router identifier 10.0.0.0, local AS number 1
Neighbor Status Codes: m - Under maintenance
  Neighbor         V  AS           MsgRcvd   MsgSent  InQ OutQ  Up/Down State   PfxRcd PfxAcc
  2.2.2.2          4  2                  7         7    0    0 00:02:43 Estab   0      0
```
```
PE1(config)#show bgp evpn summ
BGP summary information for VRF default
Router identifier 10.0.0.0, local AS number 1
Neighbor Status Codes: m - Under maintenance
  Neighbor         V  AS           MsgRcvd   MsgSent  InQ OutQ  Up/Down State   PfxRcd PfxAcc
  10.0.0.1         4  2                 24        23    0    0 00:05:52 Estab   0      0
```
```
IPv4Neighbor Capabilities:
    Multiprotocol IPv4 Unicast: advertised and received and negotiated
    Four Octet ASN: advertised and received and negotiated
    Route Refresh: advertised and received and negotiated
    Enhanced route refresh: advertised and received and negotiated
    Send End-of-RIB messages: advertised and received and negotiated
    Additional-paths recv capability:
      IPv4 Unicast: advertised
    Additional-paths send capability:
      IPv4 Unicast: received
```
```
EVPNNeighbor Capabilities:
    Multiprotocol IPv4 Unicast: advertised and received and negotiated
    Four Octet ASN: advertised and received and negotiated
    Route Refresh: advertised and received and negotiated
    Enhanced route refresh: advertised and received and negotiated
    Send End-of-RIB messages: advertised and received and negotiated
    Additional-paths recv capability:
      IPv4 Unicast: advertised
    Additional-paths send capability:
      IPv4 Unicast: received
```

**Describe alternatives you've considered**

It's liveable but the more address-families we have the more no neighbours we will have which in the end the result is simpler with the above.

**Additional context**

### LINKED ISSUE #648: EVPN-OVERLAY-PEERS is defined on Spines with Overlay Controller as EVPN Route Server
## Issue Type

- [x] Bug Report
- [x] EOS Configuration rendering error
- [ ] Documentation Issue Report

## Summary

Right now we generate EVPN_OVERLAY_PEERS peer group for `spine` even if we have `overlay-controller` defined. This should not be rendered because we don't want the `spine` to have any EVPN peers if super-spine or overlay-controller will be the route server / reflector.

## Component Name
eos_l3ls_evpn

### Expected results

```
router bgp 65100
   router-id 10.1.2.1
   no bgp default ipv4-unicast
   distance bgp 20 200 200
   graceful-restart restart-time 300
   graceful-restart
   maximum-paths 4 ecmp 4
   neighbor IPv4-UNDERLAY-PEERS peer group
   neighbor IPv4-UNDERLAY-PEERS password 7 AQQvKeimxJu+uGQ/yYvv9w==
   neighbor IPv4-UNDERLAY-PEERS send-community
   neighbor IPv4-UNDERLAY-PEERS maximum-routes 12000
```

```shell
$
```

### Actual results

```yaml
   neighbor EVPN-OVERLAY-PEERS peer group
   neighbor EVPN-OVERLAY-PEERS next-hop-unchanged
   neighbor EVPN-OVERLAY-PEERS update-source Loopback0
   neighbor EVPN-OVERLAY-PEERS bfd
   neighbor EVPN-OVERLAY-PEERS ebgp-multihop 3
   neighbor EVPN-OVERLAY-PEERS password 7 q+VNViP5i4rVjW1cxFv2wA==
   neighbor EVPN-OVERLAY-PEERS send-community
   neighbor EVPN-OVERLAY-PEERS maximum-routes 0
   neighbor IPv4-UNDERLAY-PEERS peer group
   neighbor IPv4-UNDERLAY-PEERS password 7 AQQvKeimxJu+uGQ/yYvv9w==
   neighbor IPv4-UNDERLAY-PEERS send-community
   neighbor IPv4-UNDERLAY-PEERS maximum-routes 12000
```

```shell

```

### LINKED ISSUE #649: With Overlay Controller presence the Spines still configure EVPN neighbours when not required
## Issue Type

- [x] Bug Report
- [x] EOS Configuration rendering error
- [ ] Documentation Issue Report

## Summary

With EVPN Overlay Controller knobs we want to run EVPN only between Route Servers and Leaves. With current rendering we generate EVPN peers on Spines even though we don't have any Leaf peering with them.

This does not happen with super-spine, and with Route Server / Reflector configuration we should only generate EVPN config for overlay-controller and l3leaf.

```
---
type: l3leaf

evpn_overlay_controller_groups: [ 'ROUTE_SERVERS' ]
```

Workaround is to configure empty list under Fabric so we don't generate EVPN peers

FABRIC.yml
```
# Fixes Spines to stop generating EVPN peers
evpn_overlay_controller_groups: [ ]
```

## Component Name
eos_l3ls_evpn

## Steps to reproduce

```yaml
router bgp 65100
   router-id 10.1.2.1
   no bgp default ipv4-unicast
   distance bgp 20 200 200
   graceful-restart restart-time 300
   graceful-restart
   maximum-paths 4 ecmp 4
   neighbor EVPN-OVERLAY-PEERS peer group
   neighbor EVPN-OVERLAY-PEERS next-hop-unchanged
   neighbor EVPN-OVERLAY-PEERS update-source Loopback0
   neighbor EVPN-OVERLAY-PEERS bfd
   neighbor EVPN-OVERLAY-PEERS ebgp-multihop 3
   neighbor EVPN-OVERLAY-PEERS password 7 q+VNViP5i4rVjW1cxFv2wA==
   neighbor EVPN-OVERLAY-PEERS send-community
   neighbor EVPN-OVERLAY-PEERS maximum-routes 0
   neighbor IPv4-UNDERLAY-PEERS peer group
   neighbor IPv4-UNDERLAY-PEERS password 7 AQQvKeimxJu+uGQ/yYvv9w==
   neighbor IPv4-UNDERLAY-PEERS send-community
   neighbor IPv4-UNDERLAY-PEERS maximum-routes 12000
   neighbor 10.0.1.0 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.1.0 remote-as 65001
   neighbor 10.0.1.64 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.1.64 remote-as 65001
   neighbor 10.0.3.1 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.1 remote-as 65101
   neighbor 10.0.3.5 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.5 remote-as 65101
   neighbor 10.0.3.9 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.9 remote-as 65102
   neighbor 10.0.3.13 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.13 remote-as 65102
   redistribute connected route-map RM-CONN-2-BGP
```

### Expected results

No EVPN peers on Spine

```shell
$
```

### Actual results

```shell

router bgp 65100
   router-id 10.1.2.1
   no bgp default ipv4-unicast
   distance bgp 20 200 200
   graceful-restart restart-time 300
   graceful-restart
   maximum-paths 4 ecmp 4
   neighbor EVPN-OVERLAY-PEERS peer group
   neighbor EVPN-OVERLAY-PEERS next-hop-unchanged
   neighbor EVPN-OVERLAY-PEERS update-source Loopback0
   neighbor EVPN-OVERLAY-PEERS bfd
   neighbor EVPN-OVERLAY-PEERS ebgp-multihop 3
   neighbor EVPN-OVERLAY-PEERS password 7 q+VNViP5i4rVjW1cxFv2wA==
   neighbor EVPN-OVERLAY-PEERS send-community
   neighbor EVPN-OVERLAY-PEERS maximum-routes 0
   neighbor IPv4-UNDERLAY-PEERS peer group
   neighbor IPv4-UNDERLAY-PEERS password 7 AQQvKeimxJu+uGQ/yYvv9w==
   neighbor IPv4-UNDERLAY-PEERS send-community
   neighbor IPv4-UNDERLAY-PEERS maximum-routes 12000
   neighbor 10.0.1.0 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.1.0 remote-as 65001
   neighbor 10.0.1.64 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.1.64 remote-as 65001
   neighbor 10.0.3.1 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.1 remote-as 65101
   neighbor 10.0.3.5 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.5 remote-as 65101
   neighbor 10.0.3.9 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.9 remote-as 65102
   neighbor 10.0.3.13 peer group IPv4-UNDERLAY-PEERS
   neighbor 10.0.3.13 remote-as 65102
   neighbor 10.1.2.3 peer group EVPN-OVERLAY-PEERS
   neighbor 10.1.2.3 remote-as 65101
   neighbor 10.1.2.4 peer group EVPN-OVERLAY-PEERS
   neighbor 10.1.2.4 remote-as 65101
   neighbor 10.1.2.5 peer group EVPN-OVERLAY-PEERS
   neighbor 10.1.2.5 remote-as 65102
   neighbor 10.1.2.6 peer group EVPN-OVERLAY-PEERS
   neighbor 10.1.2.6 remote-as 65102
   redistribute connected route-map RM-CONN-2-BGP

```

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes)
- [X] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes issue #649 
Fixes issue #648 
Fixes issue #373 
Part of #350 

## Component(s) name

`arista.avd.eos_l3ls_evpn`

### Refactor Overlay BGP Peerings
- Remove `evpn_overlay_controller` knob
- Remove `bgp_route_reflector` knob

Replaces with:
```yaml
#spine
      # EVPN Role for Overlay BGP Peerings | Optional, default is server
      # For IBGP overlay "server" means route-reflector. For EBGP overlay "server" means route-server.
      evpn_role: < client | server | none | default -> server  >
      # Peer with these EVPN Route Servers / Route Reflectors | Optional
      evpn_route_servers: [ < route_server_inventory_hostname >, < route_server_inventory_hostname >]

#l3leaf
      # EVPN Role for Overlay BGP Peerings | Optional, default is client
      # For IBGP overlay "server" means route-reflector. For EBGP overlay "server" means route-server.
      evpn_role: < client | server | none | default -> client >
      # Peer with these EVPN Route Servers / Route Reflectors | Optional
      evpn_route_servers: [ < route_server_inventory_hostname >, < route_server_inventory_hostname >]

#super-spine and overlay-controller
      # EVPN Role for Overlay BGP Peerings | Optional, default is none
      # For IBGP overlay "server" means route-reflector. For EBGP overlay "server" means route-server.
      evpn_role: < client | server | none | default -> none >
      # Peer with these EVPN Route Servers / Route Reflectors | Optional
      evpn_route_servers: [ < route_server_inventory_hostname >, < route_server_inventory_hostname >]
```

By default the new variables will establish regular L3LS per POD.
The variables support any combination of routing hierarchies and route-server/route-reflector placement.
Any device can be a `client` and any device can be a `server`. It is possible to configure only a one or two spines as `server` and keep the others as `none`.
`client` can peer with `server`. `server` can peer with `client` and `server`.
The `server` role ensures correct configuration of next-hop-unchanged, RTC, route-reflector-client etc. as applicable.
The `none` role means that no EVPN BGP configuration will be done on those devices. This is useful for `spine` and `super-spine`, so no unused configuration is generated if route-server functionality is placed on another device.

All templates for overlay BGP has been completely reimplemented and are now common across all device types.
Part of this means that we will have a few changes to the produced configuration.
- We will no longer have `no activate` under `address-family evpn` for peer-groups that are not relevant (redundant config)
- We will have `remote_as` per neighbor instead of on the peer-group.
- `evpn ebgp-multihop` knob will be set to a default of 3, set to a higher value to allow for larger or complex topologies.
- All bgp neighbors for evpn overlay will have a description set.

### evpn_prevent_readvertise_to_server
This then knob allows automatic configuration of route-maps on eBGP sessions towards route-servers, where the peer's ASN is filtered away. This is very useful in very large scale networks, where convergence will be quicker by not having to return all updates receives from Route-server-1 to Router-server-2 just for Route-server-2 to throw them away because of ASN path loop detection.
```yaml
evpn_prevent_readvertise_to_server : < true | false >
```

### Various bugfixes and smaller changes
Many small fixes here and there. Ex. IBGP still produced overlay peerings on spines that were not route-reflectors.
Code style updates.
Moved some variable logic to external templates to enable reuse across device types.

## How to test

This was tested with my super-spine repo. Test result can be seen in this commit - before/after refactor:

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2 [hunk 1]
```diff
@@ -54,13 +58,15 @@
 {%             else %}
 {%                 set switch.platform = l3leaf.defaults.platform %}
 {%             endif %}
-{%             if bgp_as is defined and bgp_as is not none and overlay_routing_protocol|lower == 'ibgp' %}
-{%                  set leaf.bgp_as = bgp_as %}
+{%             if bgp_as is arista.avd.defined and switch.overlay_routing_protocol == 'ibgp' %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2:61]: add `| lower` to allow various input of ibgp / iBGP etc...

`if bgp_as is arista.avd.defined and switch.overlay_routing_protocol | lower == 'ibgp' `

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2 [hunk 2]
```diff
@@ -205,9 +225,9 @@
 {%                 set leaf.evpn_rd_admin_subfield = vtep_loopback_network_summary | ipaddr('network') | ipmath(leaf.id + spine.nodes | length) %}
 {%             endif %}
 {%         elif evpn_rd_type.admin_subfield == "leaf_asn" %}
-{%             set leaf.evpn_rd_admin_subfield = leaf.bgp_as %}
+{%             set leaf.evpn_rd_admin_subfield = switch.bgp_as %}
 {%         elif evpn_rd_type.admin_subfield == "spine_asn" %}
-{%             set leaf.evpn_rd_admin_subfield = spine.bgp_as %}
+{%             set leaf.evpn_rd_admin_subfield = switch.bgp_as %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2]: I believe this should remain `spine.bgp_as`

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2 [hunk 3]
```diff
@@ -220,92 +240,26 @@
 {% if evpn_rt_type is defined and evpn_rt_type is not none %}
 {%     if evpn_rt_type.admin_subfield is defined and evpn_rt_type.admin_subfield is not none %}
 {%         if evpn_rt_type.admin_subfield == "leaf_asn" %}
-{%             set leaf.evpn_rt_admin_subfield = leaf.bgp_as %}
+{%             set leaf.evpn_rt_admin_subfield = switch.bgp_as %}
 {%         elif evpn_rt_type.admin_subfield == "spine_asn" %}
-{%             set leaf.evpn_rt_admin_subfield = spine.bgp_as %}
+{%             set leaf.evpn_rt_admin_subfield = switch.bgp_as %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2]: I believe this should remain `spine.bgp_as`

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-overlay-controller-yml.j2 [hunk 4]
```diff
@@ -47,119 +46,40 @@
...
-{%                     else %}
-{%                         set peer.bgp_as = some_switch_vars.overlay_controller.defaults.bgp_as %}
-{%                     endif %}
-{%                     set peer.ip = some_switch_vars.overlay_controller_loopback_network_summary | ipaddr('network') | ipmath(some_switch_vars.overlay_controller.nodes[some_switch].id) %}
-{%                 endif %}
-{%                 do switch.evpn_overlay_peers.append(peer) %}
+{%                 do switch.remote_switches_asn.append( remote_switch_vars.overlay_controller.nodes[some_switch].bgp_as | arista.avd.default(remote_switch_vars.overlay_controller.defaults.bgp_as)) %}
 {%             endif %}
 {%         endfor %}
+{%         if bgp_as is arista.avd.defined and switch.overlay_routing_protocol == 'ibgp' %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-overlay-controller-yml.j2:52]: add `| lower` to allow various input of ibgp / iBGP etc...

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-spine-yml.j2 [hunk 5]
```diff
@@ -8,121 +8,36 @@
...
-{%     elif underlay_routing_protocol|lower == 'ospf' %}
-{%         set switch.underlay_routing_protocol = 'ospf' %}
-{%     else %}
-{%         set switch.underlay_routing_protocol = 'ebgp' %}
-{%     endif %}
+{# Set switch.underlay_routing_protocol using external logic template #}
+{% include 'logic/switch-underlay-routing-protocol.j2' %}
+{# Set switch.overlay_routing_protocol using external logic template #}
+{% include 'logic/switch-overlay-routing-protocol.j2' %}
+{% if bgp_as is arista.avd.defined and switch.overlay_routing_protocol == 'ibgp' %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-spine-yml.j2:15]: add` | lower ` to allow various input of ibgp / iBGP etc...

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/bgp_base/ip-extcommunity-lists.j2 [hunk 6]
```diff
@@ -1,6 +1,6 @@
-{# Leaf and Spine route-maps for the underlay#}
+{# Leaf route-maps for the underlay#}
 {% if type == "l3leaf" %}
-{%     if switch.overlay_routing_protocol == "ibgp" %}
+{%     if switch.overlay_routing_protocol == "ibgp" and switch.evpn_role in ["client", "server"] %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/bgp_base/ip-extcommunity-lists.j2:3]: add | lower to allow various input of ibgp / iBGP etc...

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2 (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2]: Good catch, fixed. Thank you.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2 (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/evpn-fabric-l3leaf-yml.j2]: Good catch, fixed. Thank you.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2 [hunk 7]
```diff
@@ -0,0 +1,75 @@
+{% if switch.evpn_role == "server" %}
+{%     set switch.evpn_route_clients = {} %}
+{# #}
+{# Look for switches pointing to us as evpn_route_server #}
+{%     for some_switch in groups[fabric_name] if some_switch not in switch.evpn_route_servers | arista.avd.natural_sort %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2]: I would suggest renaming `some_switch` var to `fabric_switch`

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2 [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+{%         set some_switch_vars = hostvars[some_switch] %}
+{%         if some_switch_vars.type is arista.avd.defined('l3leaf') %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2]: I would suggest renaming some_switch_vars var to fabric_switch_vars

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2 (hunk 7 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2]: Done! Thanks.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2 (hunk 8 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/logic/switch-evpn-route-clients.j2]: Done! Thanks.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/bgp_overlay/switch-router-bgp-peer-groups-evpn.j2 [hunk 9]
```diff
@@ -0,0 +1,27 @@
+{# EBGP #}
+{% if switch.overlay_routing_protocol == "ebgp" and switch.evpn_role in ["client", "server"] %}
+    EVPN-OVERLAY-PEERS:
+      type: evpn
+      update_source: Loopback0
+      bfd: true
+      ebgp_multihop: 15
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/bgp_overlay/switch-router-bgp-peer-groups-evpn.j2]: I would suggest making `ebgp_multihop` a knob, and default to 3.

**Review by carlbuchmann — APPROVED** [review]
LGTM - please review my addition, if satisfied please merge!

**Pull request merged**